ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -192,12 +198,208 @@ static INT_PTR CALLBACK ShowMessageExtProc(HWND hwndDlg, UINT uMsg, WPARAM wPara
   return 0;
 }
 
-static INT CALLBACK GetDirectoryAltProc(HWND hwnd, UINT uMsg, LPARAM lp, LPARAM pData)
+WCHAR wstrPromptStr[4096];
+WCHAR wstrTextEntry[MAX_PATH];
+WCHAR wstrWindowStr[MAX_PATH];
```

Comment:
You're writing into this variable in every function, but you're almost never reading from it. You're reading from it in one function, and you could just be calling `c_str()` on the widened result from `window_get_caption()`, as you are doing in other places.